plugins/module_utils/ec2.py

```diff
@@ -1273,12 +1273,20 @@ def delete_egress_only_internet_gateway(client, egress_only_internet_gateway_id:
 @EC2EgressOnlyInternetGatewayErrorHandler.common_error_handler("create egress only internet gateway")
 @AWSRetry.jittered_backoff()
 def create_egress_only_internet_gateway(
-    client, vpc_id: str, tags: Optional[EC2TagSpecifications] = None
+    client, module, vpc_id: str, tags: Optional[EC2TagSpecifications] = None
 ) -> Dict[str, Any]:
-    params = {"VpcId": vpc_id}
-    if tags:
-        params["TagSpecifications"] = boto3_tag_specifications(tags, types="egress-only-internet-gateway")
-    return client.create_egress_only_internet_gateway(**params)
+    try:
+        params = {"VpcId": vpc_id}
+        if tags:
+            params["TagSpecifications"] = boto3_tag_specifications(tags, types="egress-only-internet-gateway")
+        return client.create_egress_only_internet_gateway(**params)
+    except botocore.exceptions.ClientError as e:
+        error_code = e.response["Error"]["Code"]
+        error_message = e.response["Error"]["Message"]
+        if error_code == "InvalidVpcID.NotFound":
+            module.fail_json_aws(msg=error_message, exception=f"The vpc ID '{vpc_id}' does not exist")
+        else:
+            module.fail_json_aws(msg=error_message, exception=f"AWS ClientError: {error_message}")
```

Comment:
Why not move this to where this function is called in `ec2_vpc_egress_igw.py`? That way you don't have to pass the module and all the exceptions are caught in one try/except block.

Reply:
I do not understand this change because InvalidVpcID.NotFound should be already handled here https://github.com/ansible-collections/amazon.aws/pull/2608/files#diff-11d6a927433f45aa84de2f54fa0adf33e8094ebb417c75ea476ffe61b4e4f587R1248 if I am not mistaken.